### DISCUSSION TITLE
What IS Termux? How does it work?

### DISCUSSION THREAD

**Discussion by VioletGiraffe** [termux/termux-app#3403] [Q&A] [answered]
Browsed through the docs and I don't see an explanation. Is it running its own Linux kernel on top of the Android kernel? Is it running native proceses directly in the host kernel? Is it running a full VM?

**Comment by agnostic-apollo** [comment, marked as answer]
Termux runs natively on Android (linux) kernel. No custom kernel or VM/chroot.

https://github.com/agnostic-apollo/Android-Docs/blob/master/en/docs/apps/processes/phantom-cached-and-empty-processes.md#what-are-phantom-processes

**Reply by VioletGiraffe** [reply]
Thanks a lot! I did eventually deduce from reading the issues that it's launching processes on the host kernel directly, but your link expands on that and is a very interesting read.

**Reply by agnostic-apollo** [reply]
Welcome!
